### DISCUSSION TITLE
Referencing a object inside its own initializer, is there any problem?

### DISCUSSION THREAD

**Discussion by wfelipe99** [hayes/pothos#178] [Q&A] [answered]
It's not a problem with GiraphQL and the code works as expected. Strict mode was disabled and this error was not showing up (thankfully now it's enabled, I've missed the docs on this). I had to type `ForumObject` as `any` and the problem was gone. The problem is that ts-eslint reports as "Unexpected any. Specify a different type.". Although it works and I can ignore ts-eslint, the question is: is it correctly done? Is there any way better or recommended? Can I ignore ts-eslint on this?

I have a forum which has children and its type is forum (self relation). Here's the code and the error:

```typescript
import { Forum } from '@prisma/client'
import builder from 'builder'
import db from 'utils/prisma'

const ForumObject = builder.objectRef<Forum>('Forum').implement({ //ERROR: 'ForumObject' implicitly 
//has type 'any' because it does not have a type annotation 
//and is referenced directly or indirectly in its own initializer.
  fields: (t) => ({
    ...
    children: t.field({
      type: [ForumObject], // << referencing its own
      nullable: true,
      resolve: async (parent) => await db.forum.findUnique({ where: { id: parent.id } }).children(),
    }),
    ...
```

**Comment by hayes** [comment, marked as answer]
Just split up the ref and implement into 2 statements. 
```
const ForumObject = builder.objectRef<Forum>('Forum');

ForumObject.implement({
  fields: (t) => ({
    ...
    children: t.field({
      type: [ForumObject], // << referencing its own
      nullable: true,
      resolve: async (parent) => await db.forum.findUnique({ where: { id: parent.id } }).children(),
    }),
    ...
```

**Reply by wfelipe99** [reply]
Makes sense. I still struggle with basics things. I'm waiting for the day I'm good on this hobby hahaha. Thanks!
